packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,104 +197,248 @@ class CupertinoAlertDialog extends StatelessWidget {
+                  ),
+                ],
+              ),
+            ),
+          ),
         ),
       ),
     );
   }
 }
 
-/// A button typically used in a [CupertinoAlertDialog].
-///
-/// See also:
-///
-///  * [CupertinoAlertDialog], a dialog that informs the user about situations
-///    that require acknowledgement
-class CupertinoDialogAction extends StatelessWidget {
-  /// Creates an action for an iOS-style dialog.
-  const CupertinoDialogAction({
-    this.onPressed,
-    this.isDefaultAction = false,
-    this.isDestructiveAction = false,
-    @required this.child,
-  }) : assert(child != null);
+// iOS style layout policy widget for sizing an alert dialog's content section and
+// action button section.
+//
+// The sizing policy is partially determined by whether or not action buttons
+// are stacked vertically, or positioned horizontally. [isStacked] is used to
+// indicate whether or not the buttons should be stacked vertically.
+//
+// See [_CupertinoDialogRenderBox] for specific layout policy details.
+class _CupertinoDialogRenderWidget extends MultiChildRenderObjectWidget {
+  _CupertinoDialogRenderWidget({
+    Key key,
+    @required List<Widget> children,
+    bool isStacked = false,
+  }) : _isStacked = isStacked,
+       super(key: key, children: children);
 
-  /// The callback that is called when the button is tapped or otherwise
-  /// activated.
-  ///
-  /// If this is set to null, the button will be disabled.
-  final VoidCallback onPressed;
+  final bool _isStacked;
 
-  /// Set to true if button is the default choice in the dialog.
-  ///
-  /// Default buttons are bold.
-  final bool isDefaultAction;
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _CupertinoDialogRenderBox(isStacked: _isStacked);
+  }
+}
 
-  /// Whether this action destroys an object.
-  ///
-  /// For example, an action that deletes an email is destructive.
-  final bool isDestructiveAction;
+// iOS style layout policy for sizing an alert dialog's content section and action
+// button section.
+//
+// The policy is as follows:
+//
+// If all content and buttons fit on screen:
+// The content section and action button section are sized intrinsically and centered
+// vertically on screen.
+//
+// If all content and buttons do not fit on screen:
+// A minimum height for the action button section is calculated. The action
+// button section will not be rendered shorter than this minimum.  See
+// [_CupertinoDialogActionsRenderBox] for the minimum height calculation.
+//
+// With the minimum action button section calculated, the content section is
+// laid out as tall as it wants to be, up to the point that it hits the
+// minimum button height at the bottom.
+//
+// After the content section is laid out, the action button section is allowed
+// to take up any remaining space that was not consumed by the content section.
+class _CupertinoDialogRenderBox extends RenderBox
```

Comment:
nit: we call render objects "RenderFoo" generally speaking

Reply:
done